### ISSUE TITLE
fp/no-rest-parameters false-positive

### ISSUE THREAD

**Issue by EvgenyOrekhov** [jfmengels/eslint-plugin-fp#20]
```js
function f(params) {
    const [head, ...tail] = params;
}
```

produces `Unallowed use of rest parameters. Use regular function arguments instead`,
but `params` is a regular function argument (an array).

**Comment by nickmccurdy** [comment]
It seems like this rule is catching all usages of the spread (`...`) operator, and not just rest parameters. I agree that this is a false positive though, especially since spreads are so useful for defining recursive functions over collections.

**Comment by josephpage** [comment]
```js
function f(params) {
    const { 
        head,
        ...tail
    } = params;
}
```
should also be allowed.

**Comment by joelnet** [comment]
I am getting a different false positive on this when destructing an array.

```javascript
const func = ([ head, ...tail ]) => console.log ( { head, tail })
```

**Comment by svaterlaus** [comment]
I'm also getting a false positive when destructuring objects.

```
const { page_no: pageNo = 1, ...queryParameters } = event.queryStringParameters;
```
